pkg/archive/archive.go

```diff
@@ -633,6 +647,11 @@ func TarWithOptions(srcPath string, options *TarOptions) (io.ReadCloser, error)
 				}
 
 				if err := ta.addTarFile(filePath, relFilePath); err != nil {
+					if err == errSkipFilePermissions {
+						// GH18120. Note next line is deliberately NOT logrus or %q
+						fmt.Printf("Skipping '%s' - Permission was denied\n", filePath)
```

Comment:
except you're sending it to `os.Stdout` rather than to the output stream configured by the build context, like in the code you pointed. Printing that output should be handled by the caller not by this function.